profiler/src/ProfilerEngine/Datadog.Profiler.Native.Linux/OsSpecificApi.cpp

```diff
@@ -50,8 +53,14 @@ bool GetCpuInfo(pid_t tid, bool& isRunning, uint64_t& cpuTime)
 {
     char statPath[64];
     snprintf(statPath, sizeof(statPath), "/proc/self/task/%d/stat", tid);
-    FILE* file = fopen(statPath, "r");
-    if (file == nullptr)
+
+    // load the line to be able to parse it in memory
+    std::ifstream file;
+    file.open(statPath);
```

Comment:
Previously it looks like `fopen` may possible fail `(file == nullptr)` is that no longer the case?

Reply:
> Previously it looks like `fopen` may possible fail `(file == nullptr)` is that no longer the case?

It is no more the case: the line will be empty (no exception)